sycl/source/detail/pi.cpp

```diff
@@ -370,12 +370,16 @@ bool trace(TraceLevel Level) {
 
 // Initializes all available Plugins.
 std::vector<PluginPtr> &initialize() {
-  static std::once_flag PluginsInitDone;
-  // std::call_once is blocking all other threads if a thread is already
-  // creating a vector of plugins. So, no additional lock is needed.
-  std::call_once(PluginsInitDone, [&]() {
+  // This uses static variable initialization to work around a gcc bug with
```

Comment:
@sergey-semenov, @intel/llvm-reviewers-runtime, @intel/unified-runtime-reviewers, this fix was effectively reverted by "PI removal" PR as you can see here:

https://github.com/intel/llvm/blob/683123f325df3e57309a37fa9ef2cffcdd712f80/sycl/source/detail/ur.cpp#L88-L97

I don't see that corresponding gcc issue was fixed and there is no `noexcept` specification on `initializePlugins` so my understanding that it can throw exceptions and we can still encounter the same hang. Could please someone take a look?

Reply:
It's definitely sensitive to OS & gcc version, I think originally I reproduced the problem in an Ubuntu 20.04 container.